File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/domains/Country.java
package com.chaincuerealestate.chaincuerealestate.domains;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity(name = "country")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Country {
    @Id
    private String id;
    private String name;

    public static Country create(CountryNames name) {
        return new Country(
                UUID.randomUUID().toString(),
                name.toString()
        );
    }

    public enum CountryNames {
        SWEDEN,
        SPAIN
    }
}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/domains/Country.kt
package com.example.chaincuerealestate.domains

import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.util.UUID

@Entity(name = "country")
data class Country(
    @Id
    var id: String,
    var name: String
) {
    companion object {
        fun create(name: CountryNames): Country {
            return Country(
                id = UUID.randomUUID().toString(),
                name = name.toString()
            )
        }
    }

    enum class CountryNames {
        SWEDEN,
        SPAIN
    }
}
